src/freenet/support/Fields.java

```diff
@@ -924,21 +934,33 @@ public static int compareObjectID(Object o1, Object o2) {
 		return 0;
 	}
 	
-	/** Avoid issues with overflow, 2's complement. E.g. 0-Integer.MIN_VALUE = Integer.MIN_VALUE-0. */
+	/**
+	 * Avoid issues with overflow, 2's complement. E.g. 0-Integer.MIN_VALUE = Integer.MIN_VALUE-0.
+	 * @deprecated Use {@link Integer#compare(int, int)} */
+	@Deprecated
 	public static final int compare(int x, int y) {
 		if(x > y) return 1;
 		if(y > x) return -1;
 		return 0;
 	}
 	
-	/** Avoid issues with overflow, 2's complement. */
+	/**
+	 * Avoid issues with overflow, 2's complement.
+	 * @deprecated Use {@link Long#compare(long, long)} */
+	@Deprecated
 	public static final int compare(long x, long y) {
 		if(x > y) return 1;
 		if(y > x) return -1;
 		return 0;
 	}
 
-	/** Avoid issues with NaN's. */
+	/**
+	 * Avoid issues with NaN's.
+	 * @deprecated
+	 *     Use {@link Double#compare(double, double)}. Notice: It may behave differently with
+	 *     regards to special values (e.g. -0.0). Check its source code if you need precisely
+	 *     the same behavior. */
```

Comment:
Thanks! You're absolutely right, this is dissatisfying.

I have to say sorry - In the PR, I should have pointed at the the commit message to spare you from having to write this comment, it already hinted at why I did this:
I'm not a floating point expert. So I thought before I make false claims in the JavaDoc, I better leave it vague; and instead show myself to be willing to improve it from review results in case a floating point expert shows up! :)

I aimed at the following options for resolving this:
- I intentionally committed the deprecation of different types of code (hash codes, integer stuff, float stuff) in separate commits. This is to allow you to cherry-pick the good ones and leave stuff such as this behind if you don't think it's worth the hassle.
- If someone already is a float expert and can briefly give us a hint at the behavioral differences, I'm willing to write them into a longer JavaDoc and commit it.
- I can offer to try fulfilling my duty of figuring this out myself! May take some weeks though, lots of nice stuff is waiting to be finished (WoT release!), I don't want to get stuck too much on this :)